go/merkletree2/tree.go

```diff
@@ -154,19 +174,69 @@ type PositionHashPair struct {
 	Hash     Hash     `codec:"h"`
 }
 
+type RootVersion uint8
+
+const (
+	RootVersionV1      RootVersion = 1
+	CurrentRootVersion RootVersion = RootVersionV1
+
+	RootVersionTesting RootVersion = 127
+)
+
+func (r RootVersion) GetEncodingType() EncodingType {
+	switch r {
+	case RootVersionV1:
+		return EncodingTypeBlindedSHA512_256v1
+	case RootVersionTesting:
+		return EncodingTypeForTesting
+	default:
+		panic(fmt.Sprintf("Unrecognized root version: %v", r))
+	}
+}
+
 type RootMetadata struct {
-	// TODO Add timestamp, version.....
 	_struct          struct{} `codec:",toarray"` //nolint
-	Seqno            Seqno    `codec:"n"`
-	BareRootHash     Hash     `codec:"r"`
-	SkipPointersHash Hash     `codec:"s"`
+	RootVersion      RootVersion
```

Comment:
Does the EncodingType say whether it's blinded or not?  So does that mean we might want 2 encoding types for version 1?

Reply:
EncodingType maps to an Encoder, which has a boolean to indicate whether it is a blinded encoder or not. 
We currently do not support trees that are partly blinded and partly not. I would imagine we would have RootVersionV2 for non blinded trees which would map to a non blinded encoder. Is that ok?

Reply:
I think we should incorporate in V1, since we know it is just about to happen, that we can be either blinded or non-blinded.  I'd be ok with adding an extra byte right now to convey this, I think it's worth it.